packages/server/src/core/bank_rule/specialization_groups.rs

```diff
@@ -1,71 +1,197 @@
-use crate::core::types::SpecializationGroups;
+use std::collections::{HashMap, HashSet};
+
+use crate::{
+    core::types::{SpecializationGroup, SpecializationGroups},
+    resources::course::CourseId,
+};
 
 use super::BankRuleHandler;
 
+// General comment for the whole file
+// sg = specialization_group
+// sgs = specialization_groups
+
+fn check_courses_assignment_for_sgs(
+    sgs: &[SpecializationGroup],
```

Comment:
since this is a boolean function which checks whether an assignment is valid or not, I think a better name for it would be "is_valid_assignment" or "is_valid_courses_assignment_for_sgs"
I think the 1st option is enough because we are already in the sgs context when we call this function